# HG changeset patch
# User joe
# Date 1534090615 25200
#      Sun Aug 12 09:16:55 2018 -0700
# Node ID 96552582059bc1fe22396c43b86209f087118a1f
# Parent  13a24e82d155c07f6b95795ae9c164bcca50c34c
ctrl-f switches from the command window to the file window; in a file window it cycles through the file's open windows

from kvik of #cat-v

diff -r 13a24e82d155 -r 96552582059b sys/src/cmd/samterm/main.c
--- a/sys/src/cmd/samterm/main.c	Thu Jul 12 14:21:40 2018 -0700
+++ b/sys/src/cmd/samterm/main.c	Sun Aug 12 09:16:55 2018 -0700
@@ -485,6 +485,7 @@
 	case Ksoh:
 	case Kenq:
 	case Kstx:
+	case Kack:
 		return 1;
 	}
 	return 0;
@@ -658,6 +659,17 @@
 		a = t->rasp.nrunes;
 		flsetselect(l, a, a);
 		center(l, a);
+	}else if(c == Kack){
+		int i;
+		if(work == nil)
+			return;
+		t = (Text *)work->user1;
+		i = t->front;
+		if(t->nwin>1 && which==&t->l[i])
+			do
+				i = ++i % NL;
+			while(i!=t->front && t->l[i].textfn==0);
+		current(&t->l[i]);
 	}else{
 		if(c==Kesc && typeesc>=0){
 			l->p0 = typeesc;
